### PR TITLE
made some changes to support al1 rpm builds

### PR DESCRIPTION
**Pull request by YashdalfTheGray**
### Summary

made some changes that help build rpms for AL1

### Implementation details

- added prebuilt pause containers
- normalized indentation in the `gobuild` script as well as made it align with the AL2 changes
- changed some of the make target names to also align with AL2 changes

### Testing

`make rpm`

New tests cover the changes: 

### Description for the changelog

fix: support al1 rpm builds after agent and init codebase merge

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packaging/amazon-linux-ami-integrated/ecs-agent.spec [hunk 1]
```diff
@@ -26,16 +26,15 @@ License:        Apache 2.0
...
+
+Source0:        sources.tgz
 Source1:        ecs.conf
 Source2:        ecs.service
 Source3:        amazon-ecs-volume-plugin.service
 Source4:        amazon-ecs-volume-plugin.socket
 Source5:        amazon-ecs-volume-plugin.conf
 
 BuildRequires:  golang >= 1.7
-BuildRequires:  glibc-static
```
Comment (amogh09) [inline packaging/amazon-linux-ami-integrated/ecs-agent.spec:38]: Was this dependency required only for building pause container?

File: scripts/build-agent-image [hunk 2]
```diff
@@ -36,7 +36,16 @@ cp ./misc/certs/host-certs.crt rootfs/etc/ssl/certs/ca-certificates.crt
 
 # add pause container
 mkdir -p rootfs/images/
-cp ./misc/pause-container/amazon-ecs-pause.tar rootfs/images/amazon-ecs-pause.tar
+# check for precompiled pause image tar -- if one doesn't exist,
+# you need to run make dockerfree-pause
+PAUSE_TAR_FILE="${ROOT}/misc/pause-container/pause-image-tar-files/amazon-ecs-pause.tar"
```
Comment (amogh09) [inline scripts/build-agent-image:41]: Where do we write to this path? `build-pause` seems to write to `amazon-ecs-pause-${GOARCH}.tar`?

File: scripts/gobuild.sh [hunk 3]
```diff
@@ -16,16 +16,17 @@ set -x
 set -e
 export TOPWD="$(pwd)"
 export BUILDDIR="$(mktemp -d)"
-export GOPATH="${TOPWD}/ecs-init/:${BUILDDIR}"
 export SRCPATH="${BUILDDIR}/src/github.com/aws/amazon-ecs-agent"
+export GOPATH="${TOPWD}:${BUILDDIR}"
+export GO111MODULE="auto"
```
Comment (amogh09) [inline scripts/gobuild.sh:21]: Why this change?

**Comment by amogh09** [comment]
More generally, are we adding prebuilt pause containers to optimize for speed?

File: packaging/amazon-linux-ami-integrated/ecs-agent.spec (hunk 1 above)
Comment (YashdalfTheGray) [reply packaging/amazon-linux-ami-integrated/ecs-agent.spec:38]: Yeah, it was.

File: scripts/build-agent-image (hunk 2 above)
Comment (YashdalfTheGray) [reply scripts/build-agent-image:41]: We don't specifically _write_ to this path, we have the prebuilt pause container stored at this path, so we just use that to get past the `glibc` dependency.

File: scripts/gobuild.sh (hunk 3 above)
Comment (YashdalfTheGray) [reply scripts/gobuild.sh:21]: These changes were pulled from the `scripts/gobuild.sh` from `dev` to fix a broken build. Since we keep a different AL1 (legacy) and AL2 branches, I pulled these changes in.

**Comment by YashdalfTheGray** [comment]
We've used the prebuilt pause containers as a way to get past a `glibc` dependency in the build system.

**Review by amogh09 — APPROVED** [review]

**Review by prateekchaudhry — APPROVED** [review]
